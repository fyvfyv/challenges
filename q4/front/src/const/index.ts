export const MAX_SYNOPSIS_LENGTH = 200

export const API_URL = process.env.NODE_ENV === 'development' ? 'http://localhost:3000/' : ''

export interface IBook {
  author: string
  cover: string
  rating: string
  slug: string
  synopsis: string
  title: string
  upvoted: boolean
  upvotes: number
}

export interface IBooksResponse {
  books: IBook[]
}
