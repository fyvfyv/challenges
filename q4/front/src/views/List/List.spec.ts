import { mount, createLocalVue } from '@vue/test-utils'
import Component from './index'
import Vuex, { Store } from 'vuex'
import { IBook } from '@/const'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('List', () => {
  let getters
  let store: Store<IBook[]>

  beforeEach(() => {
    getters = {
      getBooks: () => () => [
        {
          author: 'Marcel Proust',
          cover: '01.jpg',
          rating: '9.9',
          slug: 'in-search-of-lost-time',
          synopsis:
            "In Search of Lost Time, also translated as Remembrance of Things Past, novel in seven parts by Marcel Proust, published in French as À la recherche du temps perdu from 1913 to 1927. The novel is the story of Proust's own life, told as an allegorical search for truth.\nIn Search of Lost Time, also translated as Remembrance of Things Past, novel in seven parts by Marcel Proust, published in French as À la recherche du temps perdu from 1913 to 1927. The novel is the story of Proust's own life, told as an allegorical search for truth.",
          title: 'In Search of Lost Time',
          upvoted: false,
          upvotes: 1111
        },
        {
          author: 'Miguel de Cervantes',
          cover: '02.jpg',
          rating: '9.8',
          slug: 'don-quixote',
          synopsis:
            'Don Quixote is a middle-aged gentleman from the region of La Mancha in central Spain. Obsessed with the chivalrous ideals touted in books he has read, he decides to take up his lance and sword to defend the helpless and destroy the wicked.\nDon Quixote is a middle-aged gentleman from the region of La Mancha in central Spain. Obsessed with the chivalrous ideals touted in books he has read, he decides to take up his lance and sword to defend the helpless and destroy the wicked.',
          title: 'Don Quixote',
          upvoted: true,
          upvotes: 1022
        }
      ]
    }

    store = new Vuex.Store({
      getters
    })
  })

  test('is a Vue instance', () => {
    const wrapper = mount(Component, {
      stubs: ['router-link'],
      store,
      localVue
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    const wrapper = mount(Component, {
      stubs: ['router-link'],
      store,
      localVue
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
