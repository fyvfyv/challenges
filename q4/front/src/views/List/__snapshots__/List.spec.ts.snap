// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<div
  class="content"
>
  <h1
    class="title"
  >
    The books of all time
  </h1>
   
  <div
    class="wrapper"
    type="text"
  >
    <label
      for="searchField"
    >
      Search:
    </label>
     
    <input
      id="searchField"
    />
  </div>
   
  <ul
    class="books"
  >
    <li
      class="book"
    >
      <header
        class="book__headline"
      >
        <router-link-stub
          class="book__link"
          to="[object Object]"
        >
          <h2
            class="book__title"
          >
            1. In Search of Lost Time
          </h2>
        </router-link-stub>
         
        <span
          class="book__rating"
        >
          (9.9/10)
        </span>
      </header>
       
      <p
        class="book__author"
      >
        Marcel Proust
      </p>
       
      <p
        class="book__description"
      >
        In Search of Lost Time, also translated as Remembrance of Things Past, novel in seven parts by Marcel Proust, published in French as Ã€ la recherche du temps perdu from 1913 to 1927. The novel is the s...
      </p>
       
      <div
        class="book__upvotes-row"
      >
        <button
          class="book__upvote"
          type="button"
        >
          
      Upvote
    
        </button>
         
        <span>
          Upvoted 1111 times
        </span>
      </div>
       
      <router-link-stub
        class="book__image-link"
        to="[object Object]"
      >
        <img
          alt="In Search of Lost Time"
          src="01.jpg"
        />
      </router-link-stub>
    </li>
    <li
      class="book"
    >
      <header
        class="book__headline"
      >
        <router-link-stub
          class="book__link"
          to="[object Object]"
        >
          <h2
            class="book__title"
          >
            2. Don Quixote
          </h2>
        </router-link-stub>
         
        <span
          class="book__rating"
        >
          (9.8/10)
        </span>
      </header>
       
      <p
        class="book__author"
      >
        Miguel de Cervantes
      </p>
       
      <p
        class="book__description"
      >
        Don Quixote is a middle-aged gentleman from the region of La Mancha in central Spain. Obsessed with the chivalrous ideals touted in books he has read, he decides to take up his lance and sword to defe...
      </p>
       
      <div
        class="book__upvotes-row"
      >
        <button
          class="book__upvote book__upvote--checked"
          type="button"
        >
          
      Upvoted
    
        </button>
         
        <span>
          Upvoted 1022 times
        </span>
      </div>
       
      <router-link-stub
        class="book__image-link"
        to="[object Object]"
      >
        <img
          alt="Don Quixote"
          src="02.jpg"
        />
      </router-link-stub>
    </li>
  </ul>
</div>
`;
